src/app/shared/components/lazy-load/lazy-load.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, AfterViewInit, ElementRef } from '@angular/core';
+
+@Component({
+  selector: 'app-lazy-load',
+  templateUrl: './lazy-load.component.html',
+  styleUrls: ['./lazy-load.component.scss']
+})
+export class LazyLoadComponent implements AfterViewInit {
+
+  observer: IntersectionObserver;
+
+  constructor(public element: ElementRef) { }
+
+  options: any = {
+    root: null,
+    rootMargin: '0px',
+    threshold: 0.1
+  };
+
+  ngAfterViewInit() {
+    const images = this.element.nativeElement.querySelectorAll('.lazy-load');
+    this.observer = new IntersectionObserver(this.handleIntersection, this.options);
+    images.forEach(img => {
+      this.observer.observe(img);
+    });
+  }
+
+  fetchImage = (url) => {
```

Comment:
1) make private
2) specify parameter type
3) it should be declared as class method not property: 
`private fetchImage(url: String):void { ... }`